src/components/Remark.tsx

```diff
@@ -0,0 +1,91 @@
+// Copyright 2019-2020 @paritytech/bridge-ui authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+import Keyring from '@polkadot/keyring';
+import { Codec } from '@polkadot/types/types';
+import { u8aToHex } from '@polkadot/util';
+import React, { useState } from 'react';
+import { Button, Container, Input } from 'semantic-ui-react';
+import styled from 'styled-components';
+
+import { useApiSourcePromiseContext } from '../contexts/ApiPromiseSourceContext';
+import { useApiTargetPromiseContext } from '../contexts/ApiPromiseTargetContext';
+import useLoadingApi from '../hooks/useLoadingApi';
+interface Props {
+  className?: string;
+  targetChain: string;
+}
+
+const Remark = ({ className, targetChain }: Props) => {
+  const { api: sourceApi } = useApiSourcePromiseContext();
+  const { api: targetApi } = useApiTargetPromiseContext();
+
+  const areApiLoading = useLoadingApi();
+  const [remarkInput, setRemarkInput] = useState('0x');
+
+  const onChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setRemarkInput(event.target.value);
+  };
+
+  async function sendMessageRemark() {
+    const keyring = new Keyring({ type: 'sr25519' });
+    const account = keyring.addFromUri('//Alice');
+
+    const remarkCall = await targetApi.tx.system.remark(remarkInput);
+    const remarkInfo = await sourceApi.tx.system.remark(remarkInput).paymentInfo(account);
+    const weight = remarkInfo.weight.toNumber();
+
+    const call = remarkCall.toU8a();
+
+    const lane_id = new Uint8Array(8);
+    const payload = {
+      call,
+      origin: {
+        SourceAccount: account.addressRaw
+      },
+      spec_version: 1,
+      weight
+    };
+
+    // Ignoring custom types missed for TS for now.
+    // Need to apply: https://polkadot.js.org/docs/api/start/typescript.user
+    // @ts-ignore
+    const payloadType = sourceApi.registry.createType('OutboundPayload', payload);
+    // @ts-ignore
+    const messageFeeType = sourceApi.registry.createType('MessageFeeData', {
+      lane_id: '0x00000000',
+      payload: u8aToHex(payloadType.toU8a())
+    });
+
+    const estimatedFeeCall = await sourceApi.rpc.state.call<Codec>(
+      `To${targetChain}OutboundLaneApi_estimate_message_delivery_and_dispatch_fee`,
+      u8aToHex(messageFeeType.toU8a())
+    );
+
+    // @ts-ignore
+    const estimatedFeeType = sourceApi.registry.createType('Option<Balance>', estimatedFeeCall);
+    const estimatedFee = estimatedFeeType.toString();
+
+    const bridgeMessage = sourceApi.tx[`bridge${targetChain}MessageLane`].sendMessage(lane_id, payload, estimatedFee);
+
+    const nonceCall = await sourceApi.rpc.system.accountNextIndex(account.address);
+    const nonce = nonceCall.toNumber();
+    await bridgeMessage.signAndSend(account, { nonce });
+  }
+
+  if (!areApiLoading) {
+    return null;
```

Comment:
This doesn't make much sense. "If the API is NOT loading display nothing". I'm assuming  it works, so the variable name should be changed. Also `useLoadingApi` could get a rename to make it easy to understand which value you are getting (is it loading or is it NOT loading).

Reply:
I understand that the UX is not the best one, because i was looking for a quick hack to prevent the message call to be executed if the both source and target connection are not ready.  As i mentioned this is just to test the api calls and set up ... for sure UX will not have nothing to do with the current behavior.

Anyway ... i'm not getting what you are referring when you say : ` could get a rename to make it easy to understand which value you are getting)` ... i think the name `areApiLoading` is quite descriptive isn't it?

Reply:
I'm not talking about UX, only about the code and variable names. You have a variable called `areApiLoading` which I presume should be `true` if the API is loading and is `false` if the API is already connected.

In this condition you are flipping the value, so in the case of `areApiLoading == false` you are returning `null` from render. That's what doesn't make sense to me - the variables suggest that the API is no longer loading, so we can safely display something, while you are doing the opposite.

Hence I think the variable names should be renamed to explain what they are for to avoid confusion.

Reply:
ahhh yes ... sorry. I know what you mean , originally the the logic was right and after a change it got mixed. You are right